src/app/main/main.component.ts

```diff
@@ -212,21 +257,49 @@ export class MainComponent implements OnInit, OnDestroy {
       if (this._hasJobParams() && this._jobParamsValid()) {
         this.dataFormGroup.patchValue({
           sql: '/* Loading sql query from job... */',
-          projectID: this.project,
+          projectID: this.projectID,
           location: this.location
         });
-        this.dataService.getQueryFromJob(this.jobID, this.location, this.project).then((queryText) => {
+
+        this.dataService.getQueryFromJob(this.jobID, this.location, this.projectID).then((queryText) => {
           this.dataFormGroup.patchValue({
             sql: queryText.sql,
           });
         });
       } else if (this._hasTableParams() && this._tableParamsValid()) {
         this.dataFormGroup.patchValue({
-          sql: `SELECT * FROM \`${this.project}.${this.dataset}.${this.table}\`;`,
-          projectID: this.project,
+          sql: `SELECT * FROM \`${this.projectID}.${this.dataset}.${this.table}\`;`,
+          projectID: this.projectID,
         });
+      } else if (this.sharingId) {
+	this.restoreDataFromSharedStorage(this.sharingId).then((shareableValues) => {
+	  if (shareableValues) {
+	    if (shareableValues.sharingVersion != SHARING_VERSION) {
+	      throw new Error('Sharing link is invalid.');
+	    }
+	    this.dataFormGroup.patchValue({
+	      sql: '/* Loading sql query from job... */',
+	      projectID: shareableValues.projectID,
+	      location: shareableValues.location
+	    });
+	    this.dataService.getQueryFromJob(shareableValues.jobID, shareableValues.location, shareableValues.projectID).then((queryText) => {
+	      this.dataFormGroup.patchValue({
+		sql: this.convertToUserQuery(queryText.sql),
+	      });
+	      const unencryptedStyles = JSON.parse(CryptoJS.enc.Utf8.stringify(CryptoJS.AES.decrypt(shareableValues.styles, queryText.sql + queryText.bytesProcessed)));
+	      this.setNumStops(<FormGroup>this.stylesFormGroup.controls.fillColor, unencryptedStyles['fillColor'].domain.length);
+	      this.setNumStops(<FormGroup>this.stylesFormGroup.controls.fillOpacity, unencryptedStyles['fillOpacity'].domain.length);
+	      this.setNumStops(<FormGroup>this.stylesFormGroup.controls.strokeColor, unencryptedStyles['strokeColor'].domain.length);
+	      this.setNumStops(<FormGroup>this.stylesFormGroup.controls.strokeOpacity, unencryptedStyles['strokeOpacity'].domain.length);
+	      this.setNumStops(<FormGroup>this.stylesFormGroup.controls.strokeWeight, unencryptedStyles['strokeWeight'].domain.length);
+	      this.setNumStops(<FormGroup>this.stylesFormGroup.controls.circleRadius, unencryptedStyles['circleRadius'].domain.length);
+	      this.stylesFormGroup.patchValue(unencryptedStyles);
+	      this.updateStyles();
+	    }).catch((e) => this.showMessage("Cannot retrieve styling options."));
```

Comment:
This code block is pretty large for the method it's nested in, let's move it to a helper method? Indentation isn't aligned with the surrounding content, as well.

Reply:
Done.